src/main/java/it/pagopa/ecommerce/commons/documents/Transaction.java

```diff
@@ -165,21 +163,19 @@ public Transaction(
     /**
      * Primary persistence constructor
      *
-     * @param transactionId transaction unique id
-     * @param noticeCodes   notice code list
-     * @param email         user email where the payment receipt will be sent to
-     * @param status        transaction status
-     * @param origin        transaction origin
-     * @param amountTotal   transaction total amount
-     * @param feeTotal      transaction total fee
-     * @param creationDate  transaction creation date
+     * @param transactionId  transaction unique id
+     * @param paymentNotices notice code list
+     * @param email          user email where the payment receipt will be sent to
+     * @param status         transaction status
+     * @param origin         transaction origin
+     * @param feeTotal       transaction total fee
+     * @param creationDate   transaction creation date
      */
     @PersistenceConstructor
     public Transaction(
             String transactionId,
-            List<NoticeCode> noticeCodes,
-            int amountTotal,
-            int feeTotal,
+            List<PaymentNotice> paymentNotices,
+            Integer feeTotal,
```

Comment:
Proposal (given that we're already refactoring 🙂)
```suggestion
            @Nonnull Optional<Integer> feeTotal,
```

The main motivation is that when we first create the transaction we don't have fee information, so the value would be `null`

Reply:
> Proposal (given that we're already refactoring 🙂)
> 
> The main motivation is that when we first create the transaction we don't have fee information, so the value would be `null`

I will suggest to not do that modification: it will raise a sonar cube issue since optional are meant to be used only for method return types to clearly signal that a method may or may not return a value. Moreover java Optional is not serializable and that can create problems with document serialization. Would you suggest another approach?

Reply:
We could try checking whether [Jackson's support for Optional](https://github.com/FasterXML/jackson-modules-java8/tree/2.15/datatypes) is sufficient? An alternative could be something like `@Nullable Integer feeTotal`? If not in the type system I would at least add an annotation to remove ambiguity on whether `feeTotal` should be present at all times or not